tests/contrib/scrapy/test_pipeline_cerberus.py

```diff
@@ -0,0 +1,147 @@
+            id="validator is {} type, loads from path to schema".format(Item.__name__),
+        ),
+        pytest.param(
+            TestItem(),
+            {
+                SETTING_CERBERUS: {
+                    TestItem: "tests.fixtures.validators.cerberus_test_schema"
+                }
+            },
+            [assert_in_cerberus_stats(TestItem), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: [cerberus_test_schema]}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, validators are in a list repr".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            ["spidermon/validation/fields/errors/missing_required_field"],
+            id="Missing required fields",
+        ),
+    ],
+)
+def test_stats_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x in pipe.stats.stats.get_stats()
+        assert casechecker(case)
+
+
+@pytest.mark.parametrize(
+    "item,settings,cases",
+    [
+        pytest.param(
+            TestItem({"url": "example.com"}),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing simple items without errors",
+        ),
+        pytest.param(
+            TreeItem(
+                {
+                    "quotes": {"author": "Vipul Gupta", "quote": "Life is vanilla"},
+                    "child": "https://example.com",
+                }
+            ),
+            {SETTING_CERBERUS: [cerberus_tree_schema]},
+            [STATS_ITEM_ERRORS],
+            id="processing nested items without errors",
+        ),
+    ],
+)
+def test_stats_not_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x not in pipe.stats.stats.get_stats()
+        assert casechecker(case)
+
+
+def test_stats_amounts_in_pipeline():
+    item = TestItem({"title": "ScrapingHub"})
+    settings = {SETTING_CERBERUS: [cerberus_test_schema]}
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    assert pipe.stats.stats.get_stats()["spidermon/validation/validators"] is 1
+    assert (
+        pipe.stats.stats.get_stats()[
+            "spidermon/validation/fields/errors/missing_required_field"
+        ]
+        is 1
```

Comment:
`==`

Reply:
Also, it looks like `pipe.stats.stats.get_stats()` is used quite a lot.
Maybe a helper method in this test would make it easier to read

Reply:
I apologize @ejulio I think some rebasing backward might have brought it back, will get it fixed. I will replace `pipe...get_stats()` with a small function.

Reply:
I think we would like it a lot now, the helper function now forms a complete sentence.